### DISCUSSION TITLE
Invalid first argument to `register()`. ForwardRef('PredictedInstance') is not a class.

### DISCUSSION THREAD

**Discussion by gaojundong** [talmolab/sleap#966] [Help!] [answered]
Hi Team,

When I try to save the skeleton instance as a new project, a window suddenly popped up to remind me:

An error occured when attempting to save:
 Invalid first argument to `register()`. ForwardRef('PredictedInstance') is not a class.
Try saving your project with a different filename or in a different format.

I try to modify the file name as label.slp and label.v001.slp, but it still doesn't work.  this is my software version:
Software versions:
SLEAP: 1.2.8
TensorFlow: 2.9.2
Numpy: 1.22.3
Python: 3.9.13
OS: macOS-12.5.1-arm64-arm-64bit

In this software, I can save skeleton node and edges and export video, but I don't know where is the problem when I save the whole project. Can you give me some hints about this. Thank you very much.

All the best,
Gaojun

**Comment by talmo** [comment]
Hi @gaojundong,

This is a bug with the current M1 support in SLEAP which broke in the latest update. We're working it and will update you here as soon as we have a fix.

Cheers,

Talmo

**Reply by gaojundong** [reply]
Thanks for your reply. Looking forward to the latest version!

**Comment by talmo** [comment, marked as answer]
Hi @gaojundong, we just pushed a fix that should take care of this bug (#968) -- there might still be other issues we haven't run into yet, but all our tests are passing at least.

If you'd like to give it a spin, just do a `git pull` from the `sleap` directory that you cloned and try saving the project again.

**Reply by gaojundong** [reply]
Hi talmo, it works this time, thanks for your help~
